doc/Language/traits.pod6

```diff
@@ -108,6 +108,27 @@ would be a way of adding functionality which cuts across class and role
 hierarchies, or can even be used to add behaviors to independently defined
 routines.
 
+=head2 X<C<is implementation-detail>,traits> trait
+
```

Comment:
You sure about this syntax? I would say X<C<is implementation-detail>|is implementation-detail,traits> is the way I've usually seen it.

Reply:
No, I am totally not sure about this syntax.
I wonder if I am the only one who founds the indexing syntax very prone to errors, which resulted in our troubles with categories and stuff.
OTOH, I did grep the docs and saw usages of such style, so don't think it hurts.
I think writing out a clear rules explaining the indexing syntax will help a lot.

As I understand it, we have:
`X< $displayed-on-page-part | $a-text-by-that-you-can-search , $category; $another-text-for-search , $category2 ...>`
Maybe. I am not really sure. However:

```
12:=head2 X<C<lazy>|lazy (statement prefix)>
436:=head2 X<C<self>>
871:=head3 X<C<Package>, C<Module>, C<Class>, C<Role>, and C<Grammar> declaration|declarator,unit;declarator,module;declarator,package>
```
^ we have those in docs already, which sounds matching the grammar given above.

Reply:
The thing is categories are *behind* the `|` always, never close to the indexed term. Theoretically, indexing is covered by the Synopsis and should be tested by Documentable. Let's put the other one I've accepted to test and see what happens, but my money is on no.

Reply:
So do we change it? I am not sure anymore.

Reply:
Yes, lease.

Reply:
This still to go. It will fail anyway in space-after-comma.

Reply:
I would *really* prefer you to use the common syntax. This is not going to work, and it will show up in text as anything inside the `X<>`

Reply:
Apparently, my copy-pasting of your first comment somehow failed... Sorry for the noise.

Reply:
Please, select squashing as an option for this PR.